src/SharpLearning.Backend.Cntk.Test/CntkRawMnistTest.cs

```diff
@@ -0,0 +1,575 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.IO.Compression;
+using System.Net;
+using CNTK;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+// To avoid name-clash with SharpLearning.Backend.DataType.
+using CntkDataType = CNTK.DataType;
+using System.Text;
+
+namespace SharpLearning.Backend.Cntk.Test
+{
+    [TestClass]
+    public class CntkRawMnistTest
+    {
+        const string DownloadPath = "MnistTest";
+
+        [TestMethod]
+        public void MnistTrainTest()
+        {
+            // Data set
+            var mnist = Mnist.Load(DownloadPath);
+            var trainingSampleCount = mnist.TrainImages.Length;
+            var dataShape = new int[] { 28, 28, 1 };
+            var numberOfClasses = 10;
+            var outputValidationError = false;
+
+            // Set global data and device types. 
+            var dataType = CntkDataType.Float;
+            var device = DeviceDescriptor.UseDefaultDevice();
+            Trace.WriteLine("Using device: " + device.Type);
+            // These are just hardcoded in the helper class.
+            CntkLayers.DataType = dataType;
+            CntkLayers.Device = device;
+
+            // Define small convolutional neural network.
+            var net = CntkLayers.Input(dataShape);
+
+            net = CntkLayers.Conv2D(net, 3, 3, 32);
```

Comment:
Using a simple helper class CntkLayers to simplify operator construction. Each method simply wraps the needed operations to perform, convolution, Dropout and so forth.